src/QMCWaveFunctions/Fermion/DelayedUpdateCUDA.h

```diff
@@ -0,0 +1,205 @@
+//////////////////////////////////////////////////////////////////////////////////////
+// This file is distributed under the University of Illinois/NCSA Open Source License.
+// See LICENSE file in top directory for details.
+//
+// Copyright (c) 2019 QMCPACK developers.
+//
+// File developed by: Ye Luo, yeluo@anl.gov, Argonne National Laboratory
+//
+// File created by: Ye Luo, yeluo@anl.gov, Argonne National Laboratory
+//////////////////////////////////////////////////////////////////////////////////////
+
+#ifndef QMCPLUSPLUS_DELAYED_UPDATE_CUDA_H
+#define QMCPLUSPLUS_DELAYED_UPDATE_CUDA_H
+
+#include <OhmmsPETE/OhmmsVector.h>
+#include <OhmmsPETE/OhmmsMatrix.h>
+#include "CUDA/CUDAallocator.hpp"
+#include <Numerics/CUDA/cuBLAS.hpp>
+#include "QMCWaveFunctions/Fermion/delayed_update_helper.h"
+#include <cuda_runtime_api.h>
+#include "CUDA/cudaError.h"
+
+namespace qmcplusplus {
+
+  class Range
+  {
+    // [first, last) rows of Ainv
+    int first, last;
+    public:
+    Range() : first(0), last(0) { };
+    void setRange(int first_in, int last_in)
+    {
+      first = first_in;
+      last  = last_in;
+    }
+    inline void clear() { first = last = 0; };
+    inline int getOffset(int index) const
+    {
+      if(!checkRange(index)) throw std::runtime_error("index not in range \n");
+      return index-first;
+    }
+    inline bool checkRange(int index) const { return (index>=first) && (index<last); };
+  };
+
+  template<typename T>
+    struct DelayedUpdateCUDA
+    {
+      // Data staged during for delayed acceptRows
+      Matrix<T, CUDAHostAllocator<T>> U, Binv;
+      Matrix<T> V;
+      //Matrix<T> tempMat; // for debugging only
+      Matrix<T, CUDAAllocator<T>, MemorySpace::CUDA> U_gpu, V_gpu, Binv_gpu, temp_gpu, Ainv_gpu;
+      // auxiliary arrays for B
+      Vector<T> p;
+      Vector<int, CUDAHostAllocator<int>> delay_list;
+      Vector<int, CUDAAllocator<int>, MemorySpace::CUDA> delay_list_gpu;
+      int delay_count;
+
+      // the range of prefetched_Ainv_rows
+      Range prefetched_range;
+      // Ainv prefetch buffer
+      Matrix<T, CUDAHostAllocator<T>> Ainv_buffer;
+
+      // CUDA specific variables
+      cublasHandle_t handle;
+      cudaStream_t hstream;
+
+      DelayedUpdateCUDA(): delay_count(0)
+      {
+        cudaErrorCheck( cudaStreamCreate(&hstream), "cudaStreamCreate failed!" );
+        cublasErrorCheck( cublasCreate(&handle), "cublasCreate failed!" );
+        cublasErrorCheck( cublasSetStream(handle,hstream), "cublasSetStream failed!" );
+      }
+
+      ~DelayedUpdateCUDA()
+      {
+        cublasErrorCheck( cublasDestroy(handle), "cublasDestroy failed!" );
+        cudaErrorCheck( cudaStreamDestroy(hstream), "cudaStreamDestroy failed!" );
+      }
+
+      inline void resize(int norb, int delay)
```

Comment:
reset? reconfigure?

Reply:
reset and be reset values. reconfigure is not clear.
This is resize internal storage. So I still prefer resize.